src/kemal/filter_handler.cr

```diff
@@ -18,10 +18,12 @@ module Kemal
       if Kemal.config.error_handlers.has_key?(context.response.status_code)
         raise Kemal::Exceptions::CustomException.new(context)
       end
-      call_next(context)
-      call_block_for_path_type(context.request.override_method, context.request.path, :after, context)
-      call_block_for_path_type("ALL", context.request.path, :after, context)
-      context
+      unless context.get?("halted")
```

Comment:
`context.response` is closed by `halt`, so you can just test for that. And maybe also for `status_code` being present to and raise otherwise (Ithen probably an O error closed the response).

Reply:
True. It goes through `Buffered::IO#close` and `HTTP::Response::Output#unbuffered_close`, but yeah, essentially it's a noop. This should probably be changed, though. `Output` should keep track if it's closed.

Reply:
crystal-lang/crystal#6465